From aba2541482d419c01b73e973952e28c82dd540f4 Mon Sep 17 00:00:00 2001
From: Liu Jian <liujian56@huawei.com>
Date: Wed, 11 Jan 2023 09:51:00 +0800
Subject: [PATCH 1372/1398] [Huawei] kabi: fix broken kabi in struct tcp_sock

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188217

--------------------------------

This patch fix kabi problem introduced by LTS commit 96a3ddb87031
("tcp: fix tcp_cwnd_validate() to not forget is_cwnd_limited")

Considering "max_packets_seq" is used by tcp stack, so use KABI_REPLACE
to fix the kabi problem.

Signed-off-by: Liu Jian <liujian56@huawei.com>
---
 include/linux/tcp.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/tcp.h b/include/linux/tcp.h
index 74f7d4466eed..761c8033a8f8 100644
--- a/include/linux/tcp.h
+++ b/include/linux/tcp.h
@@ -269,7 +269,8 @@ struct tcp_sock {
 	u32	packets_out;	/* Packets which are "in flight"	*/
 	u32	retrans_out;	/* Retransmitted packets out		*/
 	u32	max_packets_out;  /* max packets_out in last window */
-	u32	cwnd_usage_seq;  /* right edge of cwnd usage tracking flight */
+	/* right edge of cwnd usage tracking flight */
+	KABI_REPLACE(u32 max_packets_seq, u32 cwnd_usage_seq)
 
 	u16	urg_data;	/* Saved octet of OOB data and control flags */
 	u8	ecn_flags;	/* ECN status bits.			*/
-- 
2.17.1

